recipes/zlib/1.2.11/conanfile.py

```diff
@@ -81,14 +82,14 @@ def _build_zlib_cmake(self):
         cmake.build(build_dir=".", target=make_target)
 
     def _build_zlib(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
         with tools.chdir(self._source_subfolder):
             # https://github.com/madler/zlib/issues/268
             tools.replace_in_file('gzguts.h',
                                   '#if defined(_WIN32) || defined(__CYGWIN__)',
                                   '#if defined(_WIN32) || defined(__MINGW32__)')
-            if self.settings.os == "iOS":
```

Comment:
This change is pretty old: https://github.com/conan-community/conan-zlib/commit/187c58b7e8accdb0fd660acafeceeabfa2397853

It was added by @theodelrieu. Could you explain why it's no longer required?

Reply:
I built it for simulator and arm64 + installed it on an ipad, so to me it was not needed.
maybe if I would run for tvOS or watchOS, but not iOS
@theodelrieu worked mostly or even explicit with armv7 and this arch is obsolete
on the other side, this patch does not change anything in the code, just adds a header, we can add it back.  I just wanted to know if it is required, now knowing it, I don not care but think the less patches the better

Reply:
PS: a nice solution would be to have iOS official supported and have builds for it than CI/CD could check,
btw, may I remind you on https://github.com/conan-io/conan-center-index/pull/1137 :-)  in this context

Reply:
some archeology reveals it's for https://github.com/madler/zlib/issues/62